tests/thread_float/main.c

```diff
@@ -44,39 +45,43 @@ static void timer_cb(void *arg)
     xtimer_set(&timer, OFFSET);
 }
 
-static void *thread1(void *arg)
+static void *thread1(void *_arg)
 {
-    (void) arg;
-
+    const char *arg = _arg;
     float f, init;
 
-    printf("THREAD %" PRIkernel_pid " start\n", thread_getpid());
+    mutex_lock(&lock);
+    printf("THREAD %s start\n", arg);
+    mutex_unlock(&lock);
 
-    init = 1.0 * thread_getpid();
+    /* Use number at end of thread name, e.g. 3 for "t3", to seed the calculation */
+    init = 1.0 * (arg[strlen(arg) - 1] - '0');
     f = init;
 
     while (1) {
         for (unsigned long i = 0; i < 10000ul; i++) {
             f = f + 1.0 / f;
         }
         mutex_lock(&lock);
-        printf("T(%" PRIkernel_pid "): %f\n", thread_getpid(), (double)f);
+        printf("%s: %f\n", arg, (double)f);
         mutex_unlock(&lock);
         init += 1.0;
```

Comment:
No, it should yield different results due to this. But the test will be better without :-)